### PR TITLE
Aph start edits: recipes

### PR DESCRIPTION
**Pull request by apreshill**

### CODE REVIEW

**Comment by apreshill** [comment]
Deploy preview link I want: https://deploy-preview-75--tidymodels-org.netlify.com/start/recipes/

**Comment by apreshill** [comment]
ugh hold please

File: content/start/recipes/index.Rmarkdown [hunk 1]
```diff
@@ -13,38 +13,44 @@ source(here::here("content/start/common.R"))
...
- * transforming data to be on a different scale (e.g., taking the logarithm of a variable), 
++ transforming data to be on a different scale (e.g., taking the logarithm of a variable), 
  
- * transforming whole groups of predictors together,
++ transforming whole groups of predictors together,
+
++ extracting key features from raw variables (e.g., getting the day of the week out of a date variable),
  
-and so on. This might sound an awful lot like a model formula, if you have used R's formula interface. Recipes can be used to do many of the same things, but they have a much wider range of possibilities. This guide shows how to use recipes for modeling. 
+and so on. This might sound an awful lot of steps to do within a model formula, if you have used R's formula interface. Recipes can be used to do many of the same things, but they have a much wider range of possibilities. This article shows how to use recipes for modeling.
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion
and so on. If you are familiar with R's formula interface, a lot of this might sound familiar and like what a formula already does. Recipes can be used to do many of the same things, but they have a much wider range of possibilities. This article shows how to use recipes for modeling. 
```

File: content/start/recipes/index.Rmarkdown [hunk 2]
```diff
@@ -66,96 +72,161 @@ flight_data <-
   mutate_if(is.character, as.factor)
 ```
 
-Using this estimate, about `r percent(mean(flight_data$arr_delay == "late"))` of the flights in this data set had late arrivals. A summary of the variables are:
+Let's use the [nycflights13 data](https://github.com/hadley/nycflights13) to predict whether a plane arrives more than 30 minutes late. This data set contains information on `r nrow(flight_data)` flights departing near New York City in 2013. Let's start by loading the data and making a few changes to the variables:
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion
Let's use the [nycflights13 data](https://github.com/hadley/nycflights13) to predict whether a plane arrives more than 30 minutes late. This data set contains information on `r scales::comma(nrow(flight_data))` flights departing near New York City in 2013. Let's start by loading the data and making a few changes to the variables:
```

File: content/start/recipes/index.Rmarkdown [hunk 3]
```diff
@@ -66,96 +72,161 @@ flight_data <-
...
+set.seed(555)
 # Put 3/4 of the data into the training set 
 data_split <- initial_split(flight_data, prop = 3/4)
 
 # Create data frames for the two sets:
 train_data <- training(data_split)
 test_data  <- testing(data_split)
 ```
+
+In practice, our preferred approach is to use [resampling methods](https://bookdown.org/max/FES/resampling.html) with the training set first, before using the testing set. We'll work with this single split now, then return to more advanced resampling methods in the [next article](/start/resampling/).
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion

```
I don't think we should introduce this here, because resampling is used to evaluate a model and this article only evalutes once on the test set which is fine. I think it's confusing to add it here.

File: content/start/recipes/index.Rmarkdown [hunk 4]
```diff
@@ -66,96 +72,161 @@ flight_data <-
...
 
-To get started, an initial recipe declares the roles of the columns of the data: 
+Let's initiate a new recipe: 
 
 ```{r initial-recipe}
+flights_rec <- 
+  recipe(arr_delay ~ ., data = train_data) 
+```
+
+The [`recipe()` function](https://tidymodels.github.io/recipes/reference/recipe.html) has two arguments:
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion
The [`recipe()` function](https://tidymodels.github.io/recipes/reference/recipe.html) as we used it here has two arguments:
```

Clarification, since you can use it with not-a-formula

File: content/start/recipes/index.Rmarkdown [hunk 5]
```diff
@@ -66,96 +72,161 @@ flight_data <-
...
 
-Unlike the standard model formula methods in R, a recipe **does not** automatically create dummy variables^[We don't make the assumption that dummy variables are going to be used. For one reason, [not all models require the predictors to be numeric](https://bookdown.org/max/FES/categorical-trees.html). For another reason, recipes can be used to prepare the data for other, non-model purposes that prefer factors (such as a table or plot).]. Instead, `step_dummy()` can be used for this purpose: 
+But, unlike the standard model formula methods in R, a recipe **does not** automatically create these dummy variables for you- you'll need to tell your recipe to add this step. This is for two reasons. First, many models do not require [numeric predictors](https://bookdown.org/max/FES/categorical-trees.html), so dummy variables may not always be preferred. Second, recipes can also be used for purposes outside of modeling, where non-dummy versions of the variables may work better. For example, you may want to make a table or a plot with a variable as a single factor. For those reasons, you need to explicitly tell recipes to create dummy variables using `step_dummy()`:
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]: [](url)
```suggestion
But, unlike the standard model formula methods in R, a recipe **does not** automatically create these dummy variables for you; you'll need to tell your recipe to add this step. This is for two reasons. First, many models do not require [numeric predictors](https://bookdown.org/max/FES/categorical-trees.html), so dummy variables may not always be preferred. Second, recipes can also be used for purposes outside of modeling, where non-dummy versions of the variables may work better. For example, you may want to make a table or a plot with a variable as a single factor. For those reasons, you need to explicitly tell recipes to create dummy variables using `step_dummy()`: 
```

File: content/start/recipes/index.Rmarkdown [hunk 6]
```diff
@@ -192,29 +273,29 @@ flights_rec <-
...
   set_engine("glm")
 ```
 
-How will we evaluate our model performance? Let's stick with the area under the [ROC curve](https://bookdown.org/max/FES/measuring-performance.html#class-metrics), computed using the yardstick package functions called `roc_curve()` and `roc_auc()`. 
 
-During our modeling process, we will use our recipe in multiple steps. We will: 
+We will want to use our recipe across several steps as we train and test our model. We will: 
 
- 1. **Process the recipe using the training set**: This involves any estimation or calculations on these data. For our recipe, the training set would be used to determine what dummy variable columns should be created and which zero-variance predictors are slated for removal. 
+1. **Process the recipe using the training set**: This involves any estimation or calculations based on the training set. For our recipe, the training set will be used to determine which predictors should be converted to dummy variables and which predictors will have zero-variance in the training set, and should be slated for removal.
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown:291]:
```suggestion
1. **Process the recipe using the training set**: This involves any estimation or calculations based on the training set. For our recipe, the training set will be used to determine which predictors should be converted to dummy variables and which predictors have zero variance in the training set, and should be slated for removal. 
```

File: content/start/recipes/index.Rmarkdown [hunk 7]
```diff
@@ -224,32 +305,54 @@ flights_wflow <-
...
+
+1. Trained our workflow using a single call to `fit()`. 
+
+The next step is to use the trained workflow (`flights_fit`) to predict with the unseen test data, which we will do with a single call to `predict()`. The `predict()` method applies the recipe to the new data, then passes them to the fitted model. 
+
+```{r pred-class, R.options = list(tibble.print_min = 5)}
+predict(flights_fit, test_data)
+```
+
+Because our outcome variable here is a factor, the output from `predict()` returns the predicted class: `late` versus `on_time`. But, let's say we want the predicted class probabilities for each flight instead. To return those, we can specify `type = "prob"` when we use `predict()`. We'll also bind the output with some variables from the test data too and save them together:
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion
Because our outcome variable here is a factor, the output from `predict()` returns the predicted class: `late` versus `on_time`. But, let's say we want the predicted class probabilities for each flight instead. To return those, we can specify `type = "prob"` when we use `predict()`. We'll also bind the output with some variables from the test data and save them together:
```

File: content/start/recipes/index.Rmarkdown [hunk 8]
```diff
@@ -224,32 +305,54 @@ flights_wflow <-
...
   predict(flights_fit, test_data, type = "prob") %>% 
   bind_cols(test_data %>% select(arr_delay, time_hour, flight)) 
 
 # The data look like: 
-flights_pred %>% slice(1:5)
+flights_pred
 ```
 
-We can create the ROC curve with these values, using `roc_curve()` and then piping to the `autoplot()` method: 
+Now that we have a tibble with our predicted class probabilities, how will we evaluate the performance of our workflow? We can see from these first few rows that our model predicted these 5 on time flights correctly because the values of `.pred_on_time` are *p* > .50. But we also know that we have `r nrow(flights_pred)` rows total to predict. We would like to calculate a metric that tells how well our model predicted late arrivals, compared to the true status of our outcome variable, `arr_delay`.
```
Comment (juliasilge) [inline content/start/recipes/index.Rmarkdown]:
```suggestion
Now that we have a tibble with our predicted class probabilities, how will we evaluate the performance of our workflow? We can see from these first few rows that our model predicted these 5 on time flights correctly because the values of `.pred_on_time` are *p* > .50. But we also know that we have `r scales::comma(nrow(flights_pred))` rows total to predict. We would like to calculate a metric that tells how well our model predicted late arrivals, compared to the true status of our outcome variable, `arr_delay`.
```

**Review by juliasilge — APPROVED** [review]
These are some great improvements! 💪 

I never got to give feedback on the "Build a model" changes (merged too quickly, I think) so I'm going to submit a PR.

**Comment by apreshill** [comment]
Thank you @juliasilge!
